rasa_nlu/classifiers/sklearn_intent_classifier.py

```diff
@@ -0,0 +1,138 @@
+import os
+
+from rasa_nlu.components import Component
+from rasa_nlu.training_data import TrainingData
+
+# How many intents are at max put into the output intent ranking, everything else will be cut off
+INTENT_RANKING_LENGTH = 10
+
+
+class SklearnIntentClassifier(Component):
+    """Intent classifier using the sklearn framework"""
+
+    name = "intent_classifier_sklearn"
+
+    context_provides = {
+        "process": ["intent", "intent_ranking"],
+    }
+
+    output_provides = ["intent"]
+
+    def __init__(self, clf=None, le=None):
+        """Construct a new intent classifier using the sklearn framework."""
+
+        if le is not None:
+            self.le = le
+        else:
+            from sklearn.preprocessing import LabelEncoder
+            self.le = LabelEncoder()
+        self.clf = clf
+
+    def transform_labels_str2num(self, labels):
+        # type: ([str]) -> [int]
+        """Transforms a list of strings into numeric label representation.
+
+        :param labels: List of labels to convert to numeric representation"""
+
+        return self.le.fit_transform(labels)
+
+    def transform_labels_num2str(self, y):
+        # type: ([int]) -> [str]
+        """Transforms a list of strings into numeric label representation.
+
+        :param y: List of labels to convert to numeric representation"""
+
+        return self.le.inverse_transform(y)
+
+    def train(self, training_data, intent_features, num_threads):
+        # type: (TrainingData, [float], int) -> None
+        """Train the intent classifier on a data set.
+
+        :param num_threads: number of threads used during training time"""
+        from sklearn.model_selection import GridSearchCV
+        from sklearn.svm import SVC
+
+        labels = [e["intent"] for e in training_data.intent_examples]
+
+        if len(set(labels)) < 2:
+            raise Exception("Can not train an intent classifier. Need at least 2 different classes.")
+        y = self.transform_labels_str2num(labels)
+        X = intent_features
+
+        tuned_parameters = [{'C': [1, 2, 5, 10, 20, 100], 'kernel': ['linear']}]
+        self.clf = GridSearchCV(SVC(C=1, probability=True),
```

Comment:
I like the approach in the ngram crossval of counting occurrences as a lower bound of number of folds. so something like: `cv_splits = min(10, np.min(np.bincount(y)))`